### PR TITLE
New flyway start property: quarkus.flyway.clean-at-start

### PR DESCRIPTION
**Pull request by viniciusfcf**

### CODE REVIEW

**Comment by viniciusfcf** [comment]
Related to the issue [#4232 ](url)

**Comment by gastaldi** [comment]
Looks like you have some unrelated additional commits in this PR. Can you rebase or cherry-pick just the https://github.com/quarkusio/quarkus/pull/5697/commits/bb8d861222618db86583f96e9235cfd076a7a5d6 commit? Thanks

File: integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java
Comment (gastaldi) [inline integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java]: Not sure what functionality in the extension you're testing here. Doesn't seem related to what the PR proposes.

File: extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAtStartTest.java
Comment (gastaldi) [inline extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAtStartTest.java]: This message looks wrong?

File: extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAndMigrateAtStartTest.java
Comment (gastaldi) [inline extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAndMigrateAtStartTest.java]: Move the message to the assert method below

File: extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAndMigrateAtStartTest.java
Comment (gastaldi) [inline extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAndMigrateAtStartTest.java]: Move the message to the assert method below

**Review by gastaldi — CHANGES_REQUESTED** [review]
Added some comments.

**Review by gsmet — CHANGES_REQUESTED** [review]
Let's wait for the multiple datasource support being merged before considering merging this one.

I have plans to review it early next week.

**Comment by viniciusfcf** [comment]
> Looks like you have some unrelated additional commits in this PR. Can you rebase or cherry-pick just the [bb8d861](https://github.com/quarkusio/quarkus/commit/bb8d861222618db86583f96e9235cfd076a7a5d6) commit? Thanks



> Looks like you have some unrelated additional commits in this PR. Can you rebase or cherry-pick just the [bb8d861](https://github.com/quarkusio/quarkus/commit/bb8d861222618db86583f96e9235cfd076a7a5d6) commit? Thanks

@gastaldi I think I fixed it.

File: extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAtStartTest.java
Comment (viniciusfcf) [reply extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAtStartTest.java]: Yes, removed.

File: extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAndMigrateAtStartTest.java
Comment (viniciusfcf) [reply extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAndMigrateAtStartTest.java]: Got it, Done.

File: extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAndMigrateAtStartTest.java
Comment (viniciusfcf) [reply extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionCleanAndMigrateAtStartTest.java]: Ok, done.

File: integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java
Comment (viniciusfcf) [reply integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java]: I'm not really sure about this test, tried to do it similar to Migrate test. But testing if the Flyway MigrationInfo is null, That means the history has been undone.

**Comment by gastaldi** [comment]
Sorry, but it is still not rebased properly. In your branch, do `git rebase master` and then squash the commits to a single one (`git rebase -i HEAD~2`)

**Comment by gastaldi** [comment]
Ah looks like your master branch is used in this PR. It's easier if you provide a PR in a separate branch

File: integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java
Comment (gastaldi) [reply integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java]: Right, it's just that this test seems to test the Flyway API, but you already did that in the other test case. Perhaps this can be removed

Comment (viniciusfcf) [reply integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java]: Done

File: integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java
Comment (gastaldi) [inline integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java]: Is this line break necessary?

File: integration-tests/flyway/src/main/java/io/quarkus/it/flyway/FlywayFunctionalityResource.java
Comment (gastaldi) [inline integration-tests/flyway/src/main/java/io/quarkus/it/flyway/FlywayFunctionalityResource.java]: Is this line break necessary?

Comment (viniciusfcf) [reply integration-tests/flyway/src/main/java/io/quarkus/it/flyway/FlywayFunctionalityResource.java]: I'm using [Quarkus Code Style](https://github.com/quarkusio/quarkus/blob/master/CONTRIBUTING.md) and before Commit, I run "mvn clean install" comands to format the code corretly

File: integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java
Comment (viniciusfcf) [reply integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityTest.java]: I'm using [Quarkus Code Style](https://github.com/quarkusio/quarkus/blob/master/CONTRIBUTING.md) and before Commit, I run "mvn clean install" comands to format the code corretly

**Review by gastaldi — APPROVED** [review]
LGTM, sorry for being too picky. Good job!

**Comment by viniciusfcf** [comment]
@gastaldi . Thank's your patience, it's my first PR in a open source project

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I won't have the time to get the multiple datasource support in and this one looks valuable so I'm getting this one in.

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**
